gammapy/datasets/simulate.py

```diff
@@ -576,3 +578,99 @@ def run(self, dataset, observation=None):
         geom = dataset._geom
         selection = geom.contains(events.map_coord(geom))
         return events.select_row_subset(selection)
+
+
+class SimulatedObservationMaker(MapDatasetEventSampler):
+    """
+    Sample event lists for a given observation.
+    """
+
+    def __init__(
+        self,
+        energy_axis_true,
+        energy_axis,
+        migra_axis,
+        map_width,
+        map_bin_size=0.02 * u.deg,
+        random_state="random-seed",
+    ):
+        self.random_state = get_random_state(random_state)
+        self.energy_axis_true = energy_axis_true
+        self.energy_axis = energy_axis
+        self.migra_axis = migra_axis
+        self.map_width = map_width
+        self.bin_size = map_bin_size
+
+    def run(self, observation: Observation, models=None):
+        """Run the event sampler, applying IRF corrections.
+
+        Parameters
+        ----------
+        observation : `~gammapy.data.Observation`
+            Observation to be simulated.
+        models : `~gammapy.modeling.Models`
+
+        Returns
+        -------
+        events : `~gammapy.data.EventList`
+            Event list.
+        """
+        if models is None:
+            models = Models()
+
+        pointing_icrs = observation.pointing.get_icrs(
+            observation.obstime,
+            observation.observatory_earth_location,
+        )
+        geom = WcsGeom.create(
+            skydir=pointing_icrs,
+            width=(self.map_width.to_value(u.deg), self.map_width.to_value(u.deg)),
+            binsz=self.bin_size.to_value(u.deg),
+            frame="icrs",
+            axes=[self.energy_axis],
+        )
+
+        dataset = MapDataset.create(
+            geom,
+            energy_axis_true=self.energy_axis_true,
+            migra_axis=self.migra_axis,
+            name="simulated-dataset",
+        )
+
+        components = ["exposure"]
+        if observation.edisp is not None:
+            components.append("edisp")
+        if observation.bkg is not None:
+            components.append("background")
+        if observation.psf is not None:
+            components.append("psf")
+
+        maker = MapDatasetMaker(selection=components)
+        dataset = maker.run(dataset, observation)
+        models.append(FoVBackgroundModel(dataset_name="simulated-dataset"))
+        dataset.models = models
+
+        events = None
+        if len(models) > 1:
```

Comment:
Note: while correct this logic is confusing. At this point I am asking "why only simulate stuff if we have atleast 2 models?

Ah, because 1 is always the background... there should be a better way to express this.

Reply:
Yes, the logic would support the fact that the model always has the bacgrkound. Probably not the best solution, but I don't have any smart idea now...